pagopa/io-app
# chore(IT Wallet): [SIW-1933] Handle Trustmark QR code generation failures

## Short description
This PR adds the handling of Trustmark generation failures in the Trustmark QR Code screen

## List of changes proposed in this pull request
- Added failure alert if QR Code generation fails
- Added the ability to retry the QR Code generation with an exponential backoff timer
- Updated tests

## How to test
From the details screen of a credential, tap on the Trustmark button, check that everything works as expected.

> [!TIP]
> You can make the Trustmark generation fail by mocking the `getCredentialTrustmarkActor` function and make it throw 

Ensure that when the Trustmark generation fails, the screens shows an error alert with a "Retry" button.
If you try to press the "Retry" button repeatedly, you should get an error toast telling you to wait a few seconds before trying to get the trustmark again.

## Preview
<img src="https://github.com/user-attachments/assets/fd334d54-4d61-4d9f-9fd3-89d45cfdc4ee" width="200" />
<img src="https://github.com/user-attachments/assets/8c5c729b-6368-4184-b4d8-ecf015056540" width="200" />
